src/unit_tests/remoted/test_manager.c

```diff
@@ -2411,7 +2118,1225 @@ void test_c_files(void **state)
     assert_false(multi_groups[1]->has_changed);
     assert_false(multi_groups[1]->exists);
     assert_null(multi_groups[2]);
+}
+
+void test_validate_shared_files_files_null(void **state)
+{
+    file_sum **f_sum = NULL;
+    unsigned int f_size = 0;
+
+    expect_string(__wrap_wreaddir, name, "etc/shared/test_default");
+    will_return(__wrap_wreaddir, NULL);
+
+    expect_string(__wrap__mdebug1, formatted_msg, "At validate_shared_files(): Could not open directory 'etc/shared/test_default'");
+
+    validate_shared_files("etc/shared/test_default", "test_default", "merged_tmp", &f_sum, &f_size, false, -1);
+
+    assert_non_null(groups[0]);
+    assert_string_equal(groups[0]->name, "test_default");
+    assert_null(groups[1]);
+    assert_null(f_sum);
+    assert_int_equal(f_size, 0);
+}
+
+void test_validate_shared_files_hidden_file(void **state)
+{
+    file_sum **f_sum = NULL;
+    unsigned int f_size = 0;
+    os_calloc(2, sizeof(file_sum *), f_sum);
+
+    // Initialize files structure
+    char ** files = NULL;
+    os_malloc((2) * sizeof(char *), files);
+    files[0] = strdup(".hidden_file");
+    files[1] = NULL;
+
+    expect_string(__wrap_wreaddir, name, "etc/shared/test_default");
+    will_return(__wrap_wreaddir, files);
+
+    validate_shared_files("etc/shared/test_default", "test_default", "merged_tmp", &f_sum, &f_size, false, -1);
+
+    assert_non_null(groups[0]);
+    assert_string_equal(groups[0]->name, "test_default");
+    assert_null(groups[1]);
+    assert_null(f_sum[0]);
+    assert_int_equal(f_size, 0);
+
+    free_file_sum(f_sum);
+}
+
+void test_validate_shared_files_merged_file(void **state)
+{
+    file_sum **f_sum = NULL;
+    unsigned int f_size = 0;
+    os_calloc(2, sizeof(file_sum *), f_sum);
+
+    // Initialize files structure
+    char ** files = NULL;
+    os_malloc((2) * sizeof(char *), files);
+    files[0] = strdup("merged.mg");
+    files[1] = NULL;
+
+    expect_string(__wrap_wreaddir, name, "etc/shared/test_default");
+    will_return(__wrap_wreaddir, files);
+
+    validate_shared_files("etc/shared/test_default", "test_default", "merged_tmp", &f_sum, &f_size, false, -1);
+
+    assert_non_null(groups[0]);
+    assert_string_equal(groups[0]->name, "test_default");
+    assert_null(groups[1]);
+    assert_null(f_sum[0]);
+    assert_int_equal(f_size, 0);
+
+    free_file_sum(f_sum);
+}
+
+void test_validate_shared_files_max_path_size(void **state)
+{
+    file_sum **f_sum = NULL;
+    unsigned int f_size = 0;
+    os_calloc(2, sizeof(file_sum *), f_sum);
+    const char *base_path = "190-characters-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
```

Comment:
The 190 characters string could be a define out of the function to be use for all functions that have to.